### PR TITLE
Fix code that raises new numpy deprecation warnings

### PR DESCRIPTION
**Pull request by bhazelton**
## Description

Changes:
- Fixed a bunch of places that produced new DeprecationWarnings with numpy 1.25. I think that a few of these were real bugs.
- Improve `check_warnings` to give the file and line number where warnings occur to help with debugging.
- Don't issue a warning in efield to beam when fixing auto pols to be real if imaginary values are very small (i.e. caused by numerical precision issues). We started getting a bunch of these warnings when we didn't use to.
- Made some hard to read one-line if statements in utility methods multiline. This resulted in a drop in coverage because we haven't set `branch=True` for coverage (see #1309). So I added some tests to cover all the branches.
- Made a new conda environment for python 3.11. This is needed because we can't get h5py>=3.9 to resolve with all the other dependencies in python 3.11. We need h5py>=3.9 if we have numpy>=1.25 so we don't get flooded by deprecation warnings. So in the new env for 3.11 I restricted numpy to less than 1.25. This should be reverted in the future (#1310).

## Motivation and Context

Fix CI errors due to new numpy deprecation warnings.

## Types of changes

- [x] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)
- [ ] Documentation change (documentation changes only)
- [ ] Version change
- [ ] Build or continuous integration change

## Checklist:

- [x] I have read the [contribution guide](https://github.com/RadioAstronomySoftwareGroup/pyuvdata/blob/main/.github/CONTRIBUTING.md).
- [x] My code follows the code style of this project.

Bug fix checklist:
- [ ] My fix includes a new test that breaks as a result of the bug (if possible).
- [x] All new and existing tests pass.
- [ ] I have updated the [CHANGELOG](https://github.com/RadioAstronomySoftwareGroup/pyuvdata/blob/main/CHANGELOG.md).

### CODE REVIEW

**Review by mkolopanis — COMMENTED** [review]
Looking good to me. I think my only question is why are we adding a lot of `[0]` indices in places? Did we used to have length 1 dimensions that we are collapsing?

**Comment by bhazelton** [comment]
> I think my only question is why are we adding a lot of `[0]` indices in places? Did we used to have length 1 dimensions that we are collapsing?

Yes. Numpy now complains about converting single element arrays to scalars:
`Conversion of an array with ndim > 0 to a scalar is deprecated, and will error in future. Ensure you extract a single element from your array before performing this operation. (Deprecated NumPy 1.25.)`

File: pyuvdata/tests/test_utils.py [hunk 1]
```diff
@@ -1881,18 +1882,24 @@ def test_astrometry_icrs_to_app(astrometry_args):
 
     coord_results[3] = (precalc_ra, precalc_dec)
 
+    kwargs = {}
+    extra_args = ["pm_ra", "pm_dec", "vrad", "dist"]
+    if use_extra:
+        for key in extra_args:
+            kwargs[key] = astrometry_args[key]
+    else:
+        # don't compre to precalc if not using extra arguments
```
Comment (steven-murray) [inline pyuvdata/tests/test_utils.py:1891]: typo in "compare"

**Review by steven-murray — COMMENTED** [review]
Just the one small typo

**Review by mkolopanis — APPROVED** [review]

**Review by steven-murray — APPROVED** [review]

**Pull request merged**